runtime/src/chpl-cache.c

```diff
@@ -2684,6 +2693,23 @@ struct rdcache_s* tls_cache_remote_data(void) {
   return cache;
 }
 
+// cache_lock and cache_unlock implement a "lock"
+// it's not a traditional lock because it only handles
+// the case of multiple tasks being multiplexed on one thread
+// (vs hadling separate threads accessing the same data structure).
+static
+void cache_lock(struct rdcache_s* cache) {
+  while (cache->cacheInUse != 0) {
+    chpl_task_yield();
+  }
+  cache->cacheInUse = 1;
+}
+
+static
+void cache_unlock(struct rdcache_s* cache) {
+  cache->cacheInUse = 0;
```

Comment:
I wonder if it's worth inlining these. I also wonder if it'd be a little cleaner using chpl_bool instead of int

Reply:
Eh, scratch that. Any C compiler will almost certainly decide to inline these. And bool vs int doesn't mater much. That will almost certainly not be an issue for somebody who needs to modify the cache code